velox/substrait/SubstraitToVeloxPlan.cpp

```diff
@@ -16,9 +16,72 @@
 
 #include "velox/substrait/SubstraitToVeloxPlan.h"
 #include "velox/substrait/TypeUtils.h"
+#include "velox/type/Type.h"
+#include "velox/vector/ComplexVector.h"
+#include "velox/vector/FlatVector.h"
 
 namespace facebook::velox::substrait {
 
+namespace {
+
+template <TypeKind KIND>
+VectorPtr setCellFromVariantByKind(
+
+    const std::vector<velox::variant>& value,
+    velox::memory::MemoryPool* pool,
+    std::function<bool(vector_size_t /*row*/)> isNullAt) {
+  using T = typename TypeTraits<KIND>::NativeType;
+
+  auto flatVector = std::dynamic_pointer_cast<FlatVector<T>>(
+      BaseVector::create(CppToType<T>::create(), value.size(), pool));
+
+  for (vector_size_t i = 0; i < value.size(); i++) {
+    if ((isNullAt && isNullAt(i)) || value[i].isNull()) {
+      flatVector->setNull(i, true);
+    } else {
+      flatVector->set(i, value[i].value<T>());
+    }
+  }
+
+  return flatVector;
+}
+
+template <>
+VectorPtr setCellFromVariantByKind<TypeKind::VARBINARY>(
+    const std::vector<velox::variant>& value,
+    velox::memory::MemoryPool* pool,
+    std::function<bool(vector_size_t /*row*/)> isNullAt) {
+  throw std::invalid_argument("Return of VARBINARY data is not supported");
+}
+
+template <>
+VectorPtr setCellFromVariantByKind<TypeKind::VARCHAR>(
+    const std::vector<velox::variant>& value,
+    velox::memory::MemoryPool* pool,
+    std::function<bool(vector_size_t /*row*/)> isNullAt) {
+  auto flatVector = std::dynamic_pointer_cast<FlatVector<StringView>>(
+      BaseVector::create(CppToType<StringView>::create(), value.size(), pool));
+
+  for (vector_size_t i = 0; i < value.size(); i++) {
+    if (isNullAt && isNullAt(i) || value[i].isNull()) {
+      flatVector->setNull(i, true);
+    } else {
+      flatVector->set(i, StringView(value[i].value<Varchar>()));
+    }
+  }
+  return flatVector;
+}
+
+VectorPtr setCellFromVariant(
+    const TypePtr& colType,
+    const std::vector<velox::variant>& value,
+    velox::memory::MemoryPool* pool,
+    std::function<bool(vector_size_t /*row*/)> isNullAt = nullptr) {
```

Comment:
This parameter is not needed as it is always null. Let's remove.

Reply:
remove it.